crates/pg_workspace/src/workspace/server.rs

```diff
@@ -150,13 +157,31 @@ impl WorkspaceServer {
         Ok(())
     }
 
+    fn is_ignored_by_migration_config(&self, path: &Path) -> bool {
+        let set = self.settings();
+        set.as_ref()
+            .migrations
+            .as_ref()
+            .and_then(|migration_settings| {
+                tracing::info!("Checking migration settings: {:?}", migration_settings);
+                let ignore_before = migration_settings.after.as_ref()?;
+                tracing::info!("Checking migration settings: {:?}", ignore_before);
+                let migrations_dir = migration_settings.path.as_ref()?;
+                tracing::info!("Checking migration settings: {:?}", migrations_dir);
+                let migration = migration::get_migration(path, migrations_dir)?;
+
+                Some(&migration.timestamp <= ignore_before)
+            })
+            .unwrap_or(false)
+    }
+
     /// Check whether a file is ignored in the top-level config `files.ignore`/`files.include`
     fn is_ignored(&self, path: &Path) -> bool {
         let file_name = path.file_name().and_then(|s| s.to_str());
         // Never ignore PGLSP's config file regardless `include`/`ignore`
         (file_name != Some(ConfigName::pglsp_toml())) &&
             // Apply top-level `include`/`ignore
-            (self.is_ignored_by_top_level_config(path))
+            (self.is_ignored_by_top_level_config(path) || self.is_ignored_by_migration_config(path))
```

Comment:
`is_ignored` is called by `is_path_ignored` and we add the new check here.

Reply:
very sweet!